src/pynwb/file.py

```diff
@@ -536,6 +536,8 @@ def add_electrode(self, **kwargs):
             else:
                 d.pop(col_name)  # remove args from d if not set
 
+        if 'id' in kwargs:  # was set manually by user
```

Comment:
The way that `docval` works, this always returns True. The following would be better:
```suggestion
        if kwargs['id'] is not None:  # was set manually by user
```

Example:
```python
class Test:
    @docval({'name': 'id', 'type': int, 'doc': 'the ID for the row', 'default': None})
    def add_row(self, **kwargs):
        print('id' in kwargs)

Test().add_row()  # prints True
```

Reply:
Thanks, I'll remember that one!